### ISSUE TITLE
Failed unit test with routes

### ISSUE THREAD

**Issue by brunowego** [ember-engines/ember-engines#388]
At dummy I have the mounted engine:

```js
this.mount('apollo-enterprise', { as: 'enterprise', path: '/enterprise' });
```

When I run the tests I get the follow issue:

```sh
not ok 47 PhantomJS 2.1 - Unit | Route | reports: it exists
    ---
        actual: >
            null
        expected: >
            null
        stack: >
            missingModule@http://localhost:7357/assets/vendor.js:233:93
            findModule@http://localhost:7357/assets/vendor.js:244:20
            findDeps@http://localhost:7357/assets/vendor.js:178:34
            findModule@http://localhost:7357/assets/vendor.js:248:19
            requireModule@http://localhost:7357/assets/vendor.js:31:25
            _extractDefaultExport@http://localhost:7357/assets/vendor.js:95146:27
            resolveOther@http://localhost:7357/assets/vendor.js:94862:53
            superWrapper@http://localhost:7357/assets/vendor.js:51453:27
            resolveRoute@http://localhost:7357/assets/vendor.js:16051:31
            resolve@http://localhost:7357/assets/vendor.js:15895:43
            _setupIsolatedContainer@http://localhost:7357/assets/test-support.js:7071:66
            setupContainer@http://localhost:7357/assets/test-support.js:6892:39
            http://localhost:7357/assets/test-support.js:5817:32
            initializePromise@http://localhost:7357/assets/vendor.js:69693:15
            Promise@http://localhost:7357/assets/vendor.js:70181:50
            nextStep@http://localhost:7357/assets/test-support.js:5816:54
            invokeSteps@http://localhost:7357/assets/test-support.js:5823:24
            setup@http://localhost:7357/assets/test-support.js:5770:32
            beforeEach@http://localhost:7357/assets/test-support.js:5614:34
            callHook@http://localhost:7357/assets/test-support.js:1478:26
            runHook@http://localhost:7357/assets/test-support.js:1472:15
            advance@http://localhost:7357/assets/test-support.js:1112:26
            begin@http://localhost:7357/assets/test-support.js:2762:27
            http://localhost:7357/assets/test-support.js:1970:11
        message: >
            Promise rejected before "it exists": Could not find module `apollo-enterprise/routes/reports` imported from `dummy/routes/reports`
        Log: |
    ...
ok 48 PhantomJS 2.1 - ESLint - unit/routes/reports-test.js: should pass ESLint
```

Any idea how solve it? Thanks.

**Comment by siva-sundar** [comment]
Please try loading the engine assets with a `<script>` in your tests/index.html.

**Comment by brunowego** [comment]
@siva-sundar I will close this. Thanks for the help.

**Comment by majew7** [comment]
@siva-sundar @brunowego Can you be more specific with your answer?  I have `lazyLoading: true` and this attempt did not work for me:

    <script src="{{rootURL}}engines-dist/face-member-profile/assets/engine.js"></script>
    <script src="{{rootURL}}engines-dist/face-member-profile/assets/engine-vendor.js"></script>

    <script src="/testem.js" integrity=""></script>
    <script src="{{rootURL}}assets/vendor.js"></script>
    <script src="{{rootURL}}assets/test-support.js"></script>
    <script src="{{rootURL}}assets/dummy.js"></script>
    <script src="{{rootURL}}assets/tests.js"></script>

**Comment by majew7** [comment]
Evaluating the demo repo ember-blog-engine (https://github.com/dgeb/ember-blog-engine), I observe that all 23 test pass (including `ok 23 PhantomJS 2.1 - Unit | Route | application: it exists`).

However, looking at `tests/index.html` I don't see that any new <script> tag entries were made since the creation of the repository.

I haven't found the configuration difference yet.

**Comment by majew7** [comment]
We have unit tests working now with two changes, with lazy loaded engines:
1. It's necessary to re-export pieces (routes, components, etc) in the `app` directory, for the purposes for testing.

2. 
In `tests/index.html` this order worked:

    <script src="{{rootURL}}assets/vendor.js"></script>
    <script src="/engines-dist/engine-name1/assets/engine.js"></script>
    <script src="/engines-dist/engine-name1/assets/engine-vendor.js"></script>

**Comment by amk221** [comment]
@majew7 Thanks for the technique. This works, except when run via ember exam. I raised a [separate issue](https://github.com/trentmwillis/ember-exam/issues/118) for that in case it affects anyone else.

**Comment by jbailey4** [comment]
@majew7 I was able to get my unit tests working in a _slightly_ different way, not any better IMO.

> 1. It's necessary to re-export pieces (routes, components, etc) in the app directory, for the purposes of testing.

I was able to use the `ember-engines/test-support/engine-resolver-for` module, as specified in the [README](https://github.com/ember-engines/ember-engines#unitintegration-testing-for-in-repo-engines). The README states its for in-repo-engines, but I'm using it for a stand-alone engine (not sure if that's due to some other issue I could have 😟).

Also, adding re-exports in the app directory caused module lookup issues when running the engine stand-alone and the engine's tests.

> 2. In tests/index.html this order worked:

I had to do the same.
